File: DependencyInjection/ConformingContainerAntipattern/_3_ConformingContainer/Inbound/UdpSocket.cs
using System;

namespace ConformingContainerAntipattern._3_ConformingContainer.Inbound
{
  internal interface IInputSocket
  {
    bool Receive(out byte[] frameData);
  }

  class UdpSocket : IInputSocket
  {
    public bool Receive(out byte[] frameData)
    {
      frameData = new byte[100];
      ApplicationRoot.Context.Resolve<Random>().NextBytes(frameData); //stable dependency!
      return true;
    }
  }
}

File: Java/dependency-injection/_06_conforming-container-antipattern/src/main/java/com/github/grzesiek_galezowski/ConformingContainerAntipattern/Inbound/UdpSocket.java
package com.github.grzesiek_galezowski.ConformingContainerAntipattern.Inbound;

import com.github.grzesiek_galezowski.ConformingContainerAntipattern.ApplicationRoot;

import java.util.Random;

public class UdpSocket implements InputSocket {
  public boolean receive(byte[] frameData) {
    ApplicationRoot.CONTEXT.resolve(Random.class).nextBytes(frameData); //stable dependency!
    return true;
  }
}
